src/axom/sidre/core/View.hpp

```diff
@@ -353,6 +353,19 @@ class View
    */
   int getShape(int ndims, IndexType* shape) const;
 
+  /*!
+   * \brief Return number of dimensions in data view and fill in permutation
+   *        information of this data view object.
+   *
+   *  ndims - maximum number of dimensions to return.
+   *  permutation - user supplied buffer assumed to be ndims long.
+   *
+   *  Return the number of dimensions of the view.
+   *  Return -1 if shape is too short to hold all dimensions.
+   */
+  int getPermutation(int ndims, IndexType* permutation) const;
```

Comment:
Should there also be an equivalent ``bool View::hasPermutation()`` method?

Reply:
As I've written it every `View` has a permutation. As is every `View` has an associated shape, if you don't specify the shape it's 1D. Similarly if you don't specify the permutation it's the identity.